### ISSUE TITLE
Support for Dutch 'house number extension'

### ISSUE THREAD

**Issue by casperbakker** [VIISON/AddressSplitting#7]
This is a very good package, but... :) A Dutch address can have a 'house number extension' which is officially not part of the house number. 

For example: "Moussaouistraat 22b" should be parsed like:
- street: Moussaouistraat
- house number: 22
- extension: b

Dutch parcel services require you to provide this extension separated from the house number. Would be awesome if this could be integrated into this script.

Other examples of 'extensions':
- "Moussaouistraat 22 B": "B"
- "Moussaouistraat 22-2": "-2"

**Comment by casperbakker** [comment]
For some more information, I now extended the result with the following to get what I need:

```php
<?php
        $splittedAddress = \VIISON\AddressSplitter\AddressSplitter::splitAddress($address);

        $houseNumber = $splittedAddress['houseNumber'];
        $houseNumberExtension = null;

        $housenumberMatch = preg_match("/(\d+)/", $splittedAddress['houseNumber'], $splittedHouseNumber);
        if ($housenumberMatch && strlen($splittedHouseNumber[0]) < $houseNumber)
        {
            $houseNumber = trim($splittedHouseNumber[0]);
            $houseNumberExtension = trim(substr($splittedAddress['houseNumber'], strlen($splittedHouseNumber[0])));
        }

        return [
            'street' => $splittedAddress['streetName'],
            'houseNumber' => $houseNumber,
            'houseNumberExtension' => $houseNumberExtension,
            'additionToAddress' => $splittedAddress['additionToAddress2']
        ];
```

**Comment by emilv** [comment]
Here is another failing Dutch-like address: Mannerheimintie 13ab

**Comment by svenmuennich** [comment]
Thank you for pointing this out. The problem is, that e.g. in Germany `22b` is a valid house number. Of course we can just split it, but that would be a breaking change, since the output format would change e.g. to:

```php
[
    'street' => 'Moussaouistraat',
    'houseNumber' => [
        'base' => '22',
        'extension' => 'b'
    ]
]
```

We'll discuss this internally and update you on this here.

**Comment by casperbakker** [comment]
Yeah, every country has it's own norms, very frustrating.

Would it be better (maybe in other cases as well) to also add country to the splitter? That way the splitter can split the address in certain cases differently. It would surely make this package more complex, but maybe the only way to make it support every type of address?

I know Google's phonenumber splitter also has a optional country parameter, and that does parse phone numbers better when provided.

**Comment by svenmuennich** [comment]
That's another option. I personally prefer to keep the interface simple. A breaking change in the output is not as bad as it sounds. We could even provide the complete house number too, to make migrating easier by not having to concat the two values all the time, e.g.:

```php
[
    'street' => 'Moussaouistraat',
    'houseNumber' => [
        'match' => '22b',
        'base' => '22',
        'extension' => 'b'
    ]
]
```

**Comment by casperbakker** [comment]
I like that!

**Comment by svenmuennich** [comment]
Hey @casperbakker,

we've just merged #13 which brings support for splitting the house number extension. We decided to use a _non-breaking_ result format:

```php
[
    'street' => 'Moussaouistraat',
    'houseNumber' => '22b'
    'houseNumberParts' => [
        'base' => '22',
        'extension' => 'b'
    ]
]
```

Hope this works for you 😃

**Comment by casperbakker** [comment]
That is perfect. Thanks @svenmuennich
